api/turing/api/routers_api.go

```diff
@@ -182,27 +204,45 @@ func (c RoutersController) UpdateRouter(_ *http.Request, vars RequestVars, body
 		return InternalServerError("unable to update router", err.Error())
 	}
 
+	// call webhook for router update event
+	if errWebhook := c.webhookClient.TriggerRouterEvent(ctx, webhook.OnRouterUpdated, router); errWebhook != nil {
+		log.Warnf("Error triggering webhook for event %s, router id: %d, %v",
+			webhook.OnRouterUpdated, router.ID, errWebhook)
+	}
+
 	// Deploy the new version
 	go func() {
 		err := c.deployOrRollbackRouter(project, router, routerVersion)
 		if err != nil {
 			log.Errorf("Error deploying router %s:%s:%d: %v",
 				project.Name, router.Name, routerVersion.Version, err)
 		}
+
+		// call webhook for router deployment event
+		if errWebhook := c.webhookClient.TriggerRouterEvent(ctx, webhook.OnRouterDeployed, router); errWebhook != nil {
+			log.Warnf(
+				"Error triggering webhook for event %s, router id: %d, %v",
+				webhook.OnRouterDeployed, router.ID, errWebhook,
+			)
+		}
 	}()
```

Comment:
Unlike Merlin, when Turing updates a router, it undeploys the old version and deploys the old one in a single operation. As such, we might need to insert another webhook call here for the `OnRouterUndeployed` for the old router version, preferably just before the webhook call for the `OnRouterDeployed` event. 

You might have to pay attention to the `CurrRouterVersion` [field](https://github.com/caraml-dev/turing/blob/358bea86dc00880b4ccb9256e843a2bffad073da/api/turing/models/router.go#L38) that's being sent to the webhook server (you'd want to send a request body with the old router version for the undeployment event and another request body with the new router version with the deployment event).

An additional point to take note is that you might want to log the router version ID in addition to the router ID so that's it's easier for debugging.